contracts/modules/staking/StakingPool.sol

```diff
@@ -1618,10 +1618,8 @@ contract StakingPool is IStakingPool, ERC721 {
     //              = amountOnSurge * SURGE_PRICE_RATIO * coverToCapacityRatio / 2
     //              = amountOnSurge * SURGE_PRICE_RATIO * amountOnSurge / totalCapacity / 2
 
-    uint surgePremium = amountOnSurge * SURGE_PRICE_RATIO * amountOnSurge / totalCapacity / 2;
+    uint surgePremium = amountOnSurge * amountOnSurge * NXM_PER_ALLOCATION_UNIT / totalCapacity;
```

Comment:
Note that I simplified the calculation here a bit. Left the JS implementation as it was before.

Reply:
The result of the function is the same because you removed  `* SURGE_PRICE_RATIO` which is basically `2 (1e18)` and the `/ 2` at the end which along with the allocation unit constants were "canceling each other out". The removal of the surge price ratio is a big no-no here as that is the pricing multiplier for the surge pricing. The `/ 2` that you removed is there because after calculating the area of the rectangle you divide by 2 to get the area of the triangle. If the surge price ratio was `3e18` instead, this would have failed. Pleaes revert these changes.

Reply:
Yea I can understand that it makes it less readable and maintainable. Its refactored specifically because surge price ratio is `2e18`, if we are planning to change it, we would have to bring back the division by 2. It's too bad the compiler doesn't recognize the redundant calculations here though. Would it make sense to instead just change `SURGE_PRICE_RATIO` to `2e16`, keep the div by 2, and remove the final division by 100? This way if surge price ratio changes, it should be ok?

Reply:
I agree with @shark0der  that the gains in gas are not material so i'd rather keep things clear from a mathematical point of view.

It's easier to refactor the code and update the formula and reason about it when the math matches the formula in solidity

Reply:
surprised the compiler doesn't do things here even with optimizer on

Reply:
Yea it's too bad. Anyways, I reverted the changes I made.